tests/simple-from-il-binary/simple.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright 2022 The clvk authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdio>
+#include <cstdlib>
+#include <fstream>
+#include <string.h>
+#include <vector>
+
+#define CL_TARGET_OPENCL_VERSION 120
+#include "CL/cl.h"
+#include "CL/cl_ext.h"
+
+#define BUFFER_SIZE 1024
+
+#define CHECK_CL_ERRCODE(err)                                                  \
+    do {                                                                       \
+        if (err != CL_SUCCESS) {                                               \
+            fprintf(stderr, "%s:%d error after CL call: %d\n", __FILE__,       \
+                    __LINE__, err);                                            \
+            return EXIT_FAILURE;                                               \
+        }                                                                      \
+    } while (0)
+
+int main(int argc, char* argv[]) {
+    cl_platform_id platform;
+    cl_device_id device;
+    cl_int err;
+
+    // Get the first GPU device of the first platform
+    err = clGetPlatformIDs(1, &platform, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    char platform_name[128];
+    err = clGetPlatformInfo(platform, CL_PLATFORM_NAME, sizeof(platform_name),
+                            platform_name, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    printf("Platform: %s\n", platform_name);
+
+    err = clGetDeviceIDs(platform, CL_DEVICE_TYPE_ALL, 1, &device, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    char device_name[128];
+    err = clGetDeviceInfo(device, CL_DEVICE_NAME, sizeof(device_name),
+                          device_name, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    printf("Device: %s\n", device_name);
+
+    size_t extensions_length;
+    err = clGetDeviceInfo(device, CL_DEVICE_EXTENSIONS, 0, nullptr,
+                          &extensions_length);
+    CHECK_CL_ERRCODE(err);
+
+    char* extensions = (char*)calloc(extensions_length + 1, sizeof(char));
+    err =
+        clGetDeviceInfo(device, CL_DEVICE_EXTENSIONS,
+                        sizeof(char) * extensions_length, extensions, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    if (extensions[extensions_length] != '\0') {
```

Comment:
I don't think a simple example should (have to) check this but I understand why you might have done it. The spec doesn't say anything about the string being NUL-terminated but, IIRC, this is guaranteed by the CTS. We should probably get that clarified in the specification.